delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,294 @@
+
+        To ensure options are setted on the basis of carrier_id copied from
+        Sale order or defined by default.
+
+        """
+        vals = self._values_with_carrier_options(vals)
+        return super().create(vals)
+
+    def _get_packages_from_picking(self):
+        """ Get all the packages from the picking """
+        self.ensure_one()
+        operation_obj = self.env["stock.move.line"]
+        packages = self.env["stock.quant.package"].browse()
+        operations = operation_obj.search(
+            [
+                "|",
+                ("package_id", "!=", False),
+                ("result_package_id", "!=", False),
+                ("picking_id", "=", self.id),
+            ]
+        )
+        for operation in operations:
+            # Take the destination package. If empty, the package is
+            # moved so take the source one.
+            packages |= operation.result_package_id or operation.package_id
+        return packages
+
+    def get_shipping_label_values(self, label):
+        self.ensure_one()
+        return {
+            "name": label["name"],
+            "res_id": self.id,
+            "res_model": "stock.picking",
+            "datas": label["file"],
+            "file_type": label["file_type"],
+        }
+
+    def attach_shipping_label(self, label):
+        """Attach a label returned by generate_shipping_labels to a picking"""
+        self.ensure_one()
+        data = self.get_shipping_label_values(label)
+        if label.get("package_id"):
+            data["package_id"] = label["package_id"]
+            if label.get("tracking_number"):
+                self.env["stock.quant.package"].browse(label["package_id"]).write(
+                    {"parcel_tracking": label.get("tracking_number")}
+                )
+        context_attachment = self.env.context.copy()
+        # remove default_type setted for stock_picking
+        # as it would try to define default value of attachement
+        if "default_type" in context_attachment:
+            del context_attachment["default_type"]
+        return (
+            self.env["postlogistics.shipping.label"]
+            .with_context(context_attachment)
+            .create(data)
+        )
+
+    def _set_a_default_package(self):
+        """ Pickings using this module must have a package
+            If not this method put it one silently
+        """
+        for picking in self:
+            move_lines = picking.move_line_ids.filtered(
+                lambda s: not (s.package_id or s.result_package_id)
+            )
+            if move_lines:
+                package = self.env["stock.quant.package"].create({})
+                move_lines.write({"result_package_id": package.id})
+
+    def send_to_shipper(self):
+        super().send_to_shipper()
+        if self.delivery_type == "postlogistics":
+            self.postlogistics_send_shipping()
+
+    def postlogistics_send_shipping(self):
```

Comment:
Shouldn't this implementation be in `delivery.carrier` model so it works with 
https://github.com/odoo/odoo/blob/6b363d202d9c35e8249f7bb2b0730b995fdf4a7d/addons/delivery/models/delivery_carrier.py#L173
?

Reply:
Oh, actually this function is missing :
https://github.com/OCA/delivery-carrier/blob/e518473604ec38eb416420fe0fb38227960b0f5d/delivery_carrier_label_postlogistics/models/delivery.py#L224

Reply:
Hi @TDu , you are right, actually the method implemented in picking object should be moved to delivery.carrier so that the the method will be executed when the method send_to_shipper is executed.
This will be fixed, I am discussing with jbaudoux to make some changes services & options to be implemented, and then this will be included in next update.

Reply:
fixed with the latest commit updates